importer/balanced/builder.go

```diff
@@ -211,23 +213,67 @@ func layout(db *h.DagBuilderHelper) (ipld.Node, error) {
 		if err != nil {
 			return nil, err
 		}
-		if isThereTokenMeta {
-			// Set this flag to make DAG reader side's work easier.
-			newRoot.SetTokenMeta(true)
-		}
 
 		// Fill the `newRoot` (that has the old `root` already as child)
 		// and make it the current `root` for the next iteration (when
 		// it will become "old").
-		root, fileSize, err = fillNodeRec(db, newRoot, depth)
+		root, fileSize, err = fillNodeRec(db, newRoot, depth, ft.TFile)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Add token metadata DAG, if exists, as a child to the 'newRoot'.
+	if addMetaDag {
+		root, err = attachMetadataDag(db, root, fileSize)
 		if err != nil {
 			return nil, err
 		}
 	}
 
+	if DEBUG {
```

Comment:
Not a very common way to debug, if you really need it, consider using something like: https://github.com/TRON-US/go-btfs/blob/master/cmd/btfs/main.go#L195 which is environment based.